CMake can't find `pybind11-config.cmake` file

Hi! Thank you for developing this module; it's really useful.

During installation I came across an error in finding the correct `pybind11` configuration file during set-up. For some reason, `find_package` is not able to find the right files. It's solvable by pointing the environment variable `pybind11_DIR` to the `pybind11` path that includes the `cmake` config files (e.g. `pybind11_DIR=.venv/lib/python3.8/site-packages/pybind11/share/cmake/pybind11/ pip install pydpomdp/`), but this specific example only works if `pybind11` was installed earlier. I'm using a `virtualenv`, but I was also able to reproduce this issue with a vanilla Python installation. I'm running Ubuntu on WSL2. The error is pasted below: 

```
Processing /home/maris/ellis/pomcp/pydpomdp
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: pybind11 in /home/maris/ellis/pomcp/.venv/lib/python3.8/site-packages (from pydpomdp==0.1.0) (2.6.2)
Requirement already satisfied: wheel in /home/maris/ellis/pomcp/.venv/lib/python3.8/site-packages (from pydpomdp==0.1.0) (0.37.1)
Building wheels for collected packages: pydpomdp
  Building wheel for pydpomdp (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for pydpomdp (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [70 lines of output]
      running bdist_wheel
      running build
      running build_py
      running build_ext
      CMake Error at CMakeLists.txt:4 (find_package):
        By not providing "Findpybind11.cmake" in CMAKE_MODULE_PATH this project has
        asked CMake to find a package configuration file provided by "pybind11",
        but CMake did not find one.

        Could not find a package configuration file provided by "pybind11" with any
        of the following names:

          pybind11Config.cmake
          pybind11-config.cmake

        Add the installation prefix of "pybind11" to CMAKE_PREFIX_PATH or set
        "pybind11_DIR" to a directory containing one of the above files.  If
        "pybind11" provides a separate development package or SDK, be sure it has
        been installed.


      -- Configuring incomplete, errors occurred!
      See also "/home/maris/ellis/pomcp/pydpomdp/build/temp.linux-x86_64-cpython-38/CMakeFiles/CMakeOutput.log".
      Traceback (most recent call last):
        File "/home/maris/ellis/pomcp/.venv/lib/python3.8/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 351, in <module>
          main()
        File "/home/maris/ellis/pomcp/.venv/lib/python3.8/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 333, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/home/maris/ellis/pomcp/.venv/lib/python3.8/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 249, in build_wheel
          return _build_backend().build_wheel(wheel_directory, config_settings,
        File "/tmp/pip-build-env-_2_dtpbx/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 412, in build_wheel
          return self._build_with_temp_dir(['bdist_wheel'], '.whl',
        File "/tmp/pip-build-env-_2_dtpbx/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 397, in _build_with_temp_dir
          self.run_setup()
        File "/tmp/pip-build-env-_2_dtpbx/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 335, in run_setup
          exec(code, locals())
        File "<string>", line 62, in <module>
        File "/tmp/pip-build-env-_2_dtpbx/overlay/lib/python3.8/site-packages/setuptools/__init__.py", line 87, in setup
          return distutils.core.setup(**attrs)
        File "/tmp/pip-build-env-_2_dtpbx/overlay/lib/python3.8/site-packages/setuptools/_distutils/core.py", line 185, in setup
          return run_commands(dist)
        File "/tmp/pip-build-env-_2_dtpbx/overlay/lib/python3.8/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
          dist.run_commands()
        File "/tmp/pip-build-env-_2_dtpbx/overlay/lib/python3.8/site-packages/setuptools/_distutils/dist.py", line 968, in run_commands
          self.run_command(cmd)
        File "/tmp/pip-build-env-_2_dtpbx/overlay/lib/python3.8/site-packages/setuptools/dist.py", line 1217, in run_command
          super().run_command(command)
        File "/tmp/pip-build-env-_2_dtpbx/overlay/lib/python3.8/site-packages/setuptools/_distutils/dist.py", line 987, in run_command
          cmd_obj.run()
        File "/tmp/pip-build-env-_2_dtpbx/overlay/lib/python3.8/site-packages/wheel/bdist_wheel.py", line 299, in run
          self.run_command('build')
        File "/tmp/pip-build-env-_2_dtpbx/overlay/lib/python3.8/site-packages/setuptools/_distutils/cmd.py", line 319, in run_command
          self.distribution.run_command(command)
        File "/tmp/pip-build-env-_2_dtpbx/overlay/lib/python3.8/site-packages/setuptools/dist.py", line 1217, in run_command
          super().run_command(command)
        File "/tmp/pip-build-env-_2_dtpbx/overlay/lib/python3.8/site-packages/setuptools/_distutils/dist.py", line 987, in run_command
          cmd_obj.run()
        File "/tmp/pip-build-env-_2_dtpbx/overlay/lib/python3.8/site-packages/setuptools/_distutils/command/build.py", line 132, in run
          self.run_command(cmd_name)
        File "/tmp/pip-build-env-_2_dtpbx/overlay/lib/python3.8/site-packages/setuptools/_distutils/cmd.py", line 319, in run_command
          self.distribution.run_command(command)
        File "/tmp/pip-build-env-_2_dtpbx/overlay/lib/python3.8/site-packages/setuptools/dist.py", line 1217, in run_command
          super().run_command(command)
        File "/tmp/pip-build-env-_2_dtpbx/overlay/lib/python3.8/site-packages/setuptools/_distutils/dist.py", line 987, in run_command
          cmd_obj.run()
        File "<string>", line 31, in run
        File "<string>", line 59, in build_extension
        File "/usr/lib/python3.8/subprocess.py", line 364, in check_call
          raise CalledProcessError(retcode, cmd)
      subprocess.CalledProcessError: Command '['cmake', '/home/maris/ellis/pomcp/pydpomdp/pydpomdp', '-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=/home/maris/ellis/pomcp/pydpomdp/build/lib.linux-x86_64-cpython-38/', '-DPYTHON_EXECUTABLE=/home/maris/ellis/pomcp/.venv/bin/python3', '-DCMAKE_BUILD_TYPE=Release']' returned non-zero exit status 1.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pydpomdp
Failed to build pydpomdp
ERROR: Could not build wheels for pydpomdp, which is required to install pyproject.toml-based projects
```

I think there's multiple solutions to this problem, one of which I've added as diff patch below (can't open a pull request since I'm not allowed to push). Feel free to ignore it, as I don't have a lot of experience with building Python modules; it was a fun way for me to see what would be a workable solution for this problem. Appending the module installation path to the prefix path might also work ([CMAKE_PREFIX_PATH](https://cmake.org/cmake/help/latest/variable/CMAKE_PREFIX_PATH.html)).

I think another possible solution would be to use [FindPython](https://cmake.org/cmake/help/latest/module/FindPython.html), but `CMake>=3.12` would be required for this.

```
diff --git a/setup.py b/setup.py
index 4401a7d..338ab4b 100644
--- a/setup.py
+++ b/setup.py
@@ -36,8 +36,16 @@ class CMakeBuild(build_ext):
         if not extdir.endswith(os.path.sep):
             extdir += os.path.sep

+        try:
+            import pybind11
+        except ImportError:
+            print("Successful installation of ``pydpomdp'' requires ``pybind11''. Exiting.")
+            exit(-1)
+
         cmake_args = ['-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=' + extdir,
-                      '-DPYTHON_EXECUTABLE=' + sys.executable]
+                      '-DPYTHON_EXECUTABLE=' + sys.executable,
+                      '-Dpybind11_DIR=' + pybind11.get_cmake_dir()
+                     ]

         cfg = 'Debug' if self.debug else 'Release'
         build_args = ['--config', cfg]
 ```
After applying the above patch, the installation instructions as in the README work out of the box.

--- a/setup.py
+++ b/setup.py
@@ -36,8 +36,16 @@ class CMakeBuild(build_ext):
         if not extdir.endswith(os.path.sep):
             extdir += os.path.sep
 
+        try:
+            import pybind11
+        except ImportError:
+            print("Successful installation of ``pydpomdp'' requires ``pybind11''. Exiting.")
+            exit(-1)
+
         cmake_args = ['-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=' + extdir,
-                      '-DPYTHON_EXECUTABLE=' + sys.executable]
+                      '-DPYTHON_EXECUTABLE=' + sys.executable,
+                      '-Dpybind11_DIR=' + pybind11.get_cmake_dir()
+                     ]
 
         cfg = 'Debug' if self.debug else 'Release'
         build_args = ['--config', cfg]

